ocaml/xapi/xapi_vm.ml

```diff
@@ -422,6 +422,15 @@ let power_state_reset ~__context ~vm =
 let suspend ~__context ~vm =
   Db.VM.set_ha_always_run ~__context ~self:vm ~value:false;
   debug "Setting ha_always_run on vm=%s as false during VM.suspend" (Ref.string_of vm);
+  (* set compatibility meta data for all NVIDIA vGPUs *)
```

Comment:
I'd make this an helper in the `Xapi_gpumon` module as it is for `assert_pgpu_is_compatible_with_vm` and expose as little logic as we can outside that module